src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw

```diff
@@ -1250,4 +1250,19 @@ Please specify one of them using the `--source` option to proceed.</value>
   <data name="InstallArchitectureArgumentDescription" xml:space="preserve">
     <value>Select the architecture to install</value>
   </data>
+  <data name="IncludeUnknownArgumentDescription" xml:space="preserve">
+    <value>Upgrade packages even if their current version cannot be determined</value>
+  </data>
+  <data name="UpgradeUnknownVersionExplanation" xml:space="preserve">
+    <value>This package's version number cannot be determined. To upgrade it anyway, add the argument --include-unknown to your previous command.</value>
+    <comment>{Locked="--include-unknown"}</comment>
+  </data>
+  <data name="UpgradeUnknownCount" xml:space="preserve">
+    <value>package's version numbers cannot be determined. Use "--include-unknown" to see all results.</value>
+    <comment>{Locked="--include-unknown"}</comment>
+  </data>
+  <data name="UpgradeUnknownCountSingle" xml:space="preserve">
+    <value>package's version number cannot be determined. Use "--include-unknown" to see all results.</value>

```

Comment:
I don't think this formatting ([n] + " package's version number") works for all languages. For example, in Spanish I would probably say this as "El número de versión de [n] paquetes no pudo ser determinada.", which doesn't have the number at the beginning. You can probably play with the phrasing to make it work (e.g. "[n] paquetes tienen números de versión que no pudieron ser determinados"), but I'm not sure if you can do that in all languages. At least we should add a comment about how this string is used so that the localization team can make it work.

Reply:
We have a general problem with this, and a bug from the localization team to prove it 😄 The right thing to do is actually to include format specifiers (not formally defined, but think like C#'s `{0}`) to our strings and replace them at runtime with the actual values.  I wanted it to be a generalized approach when we fix it, hopefully making it easy at the call sites to use consistent syntax.

Reply:
What if the message were more general?
```
Some packages' versions could not be determined. Use "--include-unknown" to see all results"
```
That would be easily localizable and avoid having to count the number of packages, saving on processing time and memory for systems with large numbers of packages installed

Reply:
The count was added after a request from @denelon: https://github.com/microsoft/winget-cli/pull/1765#issuecomment-985060905

Reply:
I wasn’t sure if denelon meant more that it would be helpful to have a message (since looking through the commits, it seems like it was only if you tried to upgrade a specific package?) or if he specifically meant to have a count

Reply:
The design he gave left a spot for a count (`.... <n> have unknown version...`).

There are quite a few spots where the format specifier solution will need to be used (where strings are broken up into pieces to fit the localized bits, like in [`ReportIdentity`](https://github.com/microsoft/winget-cli/blob/60db65945217445651b8139d264d1326e9c23b54/src/AppInstallerCLICore/Workflows/WorkflowBase.cpp#L39)). I think the phrasing change should be okay for now (I made it more verbose based on this review and added a comment for localization).

Reply:
In either case, a message should be displayed, and the count is better.